src/js/components/AppPageComponent.jsx

```diff
@@ -439,12 +440,14 @@ var AppPageComponent = React.createClass({
       content = this.getTaskDetailComponent();
     }
 
+    var group = ViewHelper.getGroupFromAppId(state.appId);
+
     return (
       <div>
-        <AppBreadcrumbsComponent
-          activeTaskId={state.activeTaskId}
-          activeViewIndex={state.activeViewIndex}
-          appId={state.appId} />
+        <BreadcrumbComponent
+          group={group}
+          app={state.appId}
```

Comment:
Yeah, I think you're probably right. I did consider this whilst implementing. My logic was that this component depends entirely on strings in its `props` and never has to distinguish between an object and its array, so I used the shorter form.